packages/export-tools/src/find-unused-exports.mjs

```diff
@@ -0,0 +1,118 @@
+import { Project } from 'ts-morph';
+import { writeFile } from './utils.mjs';
+
+// 1. Locate ui-components, ui-premium, and labkey home via env-vars (consult labkey build code)
+// 2. Find all exports from ui-components, ui-premium (use ts-morph)
+// 3. Find all imports of ui-components and ui-premium in known locations based on env-vars (use ts-morph)
+// 4. Diff the exports and imports, output the diffs (one file per package)
+// 5. Consider using ts-morph to modify the AST and remove the unused exports (consult ts-morph docs)
+//     - Only do this when --write flag is present, don't output diff files when write is true
+//     - Bonus points: consider using ts-morph to export everything in sorted order.
+
+// S-1, S-999
+// S-1, S-2
+
+if (process.env.LABKEY_HOME === undefined) {
+    throw new Error('Missing env var LABKEY_HOME');
+}
+
+if (process.env.LABKEY_UI_COMPONENTS_HOME === undefined) {
+    throw new Error('Missing env var LABKEY_UI_COMPONENTS_HOME');
+}
+
+if (process.env.LABKEY_UI_PREMIUM_HOME === undefined) {
+    throw new Error('Missing env var LABKEY_UI_PREMIUM_HOME');
+}
+
+const components = `${process.env.LABKEY_UI_COMPONENTS_HOME}packages/components/src`;
+const premium = `${process.env.LABKEY_UI_PREMIUM_HOME}src`;
+const modules = `${process.env.LABKEY_HOME}server/modules`
+const limsModules = `${modules}/limsModules`;
+const biologics = `${limsModules}/biologics`;
+const sampleManagement = `${limsModules}/sampleManagement`;
+const inventory = `${limsModules}/inventory`;
+const puppeteer = `${limsModules}/puppeteer`;
+const labbook = `${limsModules}/labbook`;
+const platform = `${modules}/platform`;
+const core = `${platform}/core`;
+const assay = `${platform}/assay`;
+const experiment = `${platform}/experiment`;
+const pipeline = `${platform}/pipeline`;
+const moduleEditor = `${modules}/premiumModules/moduleEditor`;
+const reactExamples = `${modules}/tutorialModules/reactExamples`;
+// TODO: there are probably more modules using ui-components that I am missing
```

Comment:
my local text search in IntelliJ includes the following as well:
- commonAssays/elisa
- premiumModules/ontology
- premiumModules/provenance
- tutorialModules/reactExamples
- ehrModules/ehr
- ehrModules/EHR_App
- ehrModules/labkey-ui-ehr
- snprcEHRModules/snprc_ehr
- wnprc-modules/WNPRC_Purchasing

Reply:
Ok, I added these additional modules, with the exception of provenance because we are actively not updating that module (at least for the time being).